js/components/message.jsx

```diff
@@ -1,14 +1,29 @@
 var React = require('react');
+var TimePresenter = require('../utils/time_presenter');
 
 var Message = React.createClass({
   render: function() {
+    var message = this.props.message;
+    var displayTime = TimePresenter.presentMessageTime(message.timestamp, this.state.currentTime);
     return (
-      <div className='message'>
-        [{this.props.message.timestamp}]
-        {this.props.message.user}:
-        {this.props.message.text}
+      <div className="message">
+        <div className="sender-stamp">
+          <div className="message-sender">{message.user}</div>
+          <div className="message-time">{displayTime}</div>
+        </div>
+        <div className="message-text">{message.text}</div>
       </div>
     );
+  },
+
+  getInitialState: function() {
+    return { currentTime: new Date() };
+  },
+
+  componentDidMount: function() {
+    setInterval(function() {
+      this.setState({currentTime: new Date()});
+    }.bind(this), 60000);
   }
```

Comment:
hm.. possible to rerender the entire list of messages once per minute? vs. having like 100 intervals set at once.. also since the messages come in at staggered times, it seems like there would be timestamps changing every few seconds, which would drive me nuts.